src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -109,16 +123,24 @@ MultiRouterPlanCreate(Query *query, MultiExecutorType taskExecutorType)
 	{
 		ErrorIfModifyQueryNotSupported(query);
 		task = RouterModifyTask(query);
+		jobQuery = query;
```

Comment:
That require us that we would evaluate evaluate actual values of const functions  since event const values may come wrapped inside function calls in original query. But I think we can take similar action as we did for select tasks.